source/D2Common/src/D2StatList.cpp

```diff
@@ -66,89 +175,74 @@ int __fastcall sub_6FDB5830(D2StatListExStrc* pStatListEx, int nLayer_StatId)
 		return 0;
 	}
 
-	nValue = sub_6FDB6340(pStatListEx, nLayer_StatId, 0);
+	nAccumulatedValue = STATLIST_GetBaseStat_6FDB6340(pStatListEx, nLayer_StatId, 0);
 
-	if (pStatListEx->dwFlags & 0x80000000)
+	if (STATLIST_IsExtended(pStatListEx))
 	{
 		for (D2StatListStrc* i = pStatListEx->pMyLastList; i; i = i->pPrevLink)
 		{
 			if (!(pItemStatCostTxtRecord->dwItemStatFlags & gdwBitMasks[ITEMSTATCOSTFLAGINDEX_DAMAGERELATED]) || !(i->dwFlags & 0x40000000))
 			{
-				nValue += sub_6FDB63E0((D2StatListExStrc*)i, nLayer_StatId, 0);
+				nAccumulatedValue += STATLIST_GetTotalStat_6FDB63E0((D2StatListExStrc*)i, nLayer_StatId, 0);
 			}
 		}
 	}
 
 	if (!pItemStatCostTxtRecord->unk0x51[1])
 	{
-		return nValue;
+		return nAccumulatedValue;
 	}
 
-	nOldValue = nValue;
+	nPreviousOpStatValue = nAccumulatedValue;
 
 	for (int i = 0; i < ARRAY_SIZE(pItemStatCostTxtRecord->pOpStatData); ++i)
 	{
-		switch (pItemStatCostTxtRecord->pOpStatData[i].nOp)
+		D2OpStatDataStrc& opStatData = pItemStatCostTxtRecord->pOpStatData[i];
+		switch (opStatData.nOp)
 		{
 		case 0:
-			return nValue;
+			return nAccumulatedValue;
 
 		case 1:
+
 			if (pStatListEx->pOwner)
 			{
 				if (pStatListEx->pOwner->dwUnitType == UNIT_ITEM)
 				{
-					if (pStatListEx->pOwner->pStatListEx)
-					{
-						nOldValue = sub_6FDB6340(pStatListEx->pOwner->pStatListEx, nStatId << 16, pItemStatCostTxtRecord);
-					}
-					else
+					nPreviousOpStatValue = 0;
+					if (pItemStatCostTxtRecord && pStatListEx->pOwner->pStatListEx)
 					{
-						nOldValue = 0;
+						nPreviousOpStatValue = STATLIST_GetBaseStat_6FDB6340(pStatListEx->pOwner->pStatListEx, nStatId << 16, pItemStatCostTxtRecord);
 					}
 				}
 
-				if (nOldValue)
+				if (nPreviousOpStatValue)
 				{
-					nTemp = sub_6FDB5830(pStatListEx, pItemStatCostTxtRecord->pOpStatData[i].nStat << 16);
-					if (nTemp)
-					{
-						nValue += nOldValue * nTemp / 100;
-					}
+					return ComputeStatPercentage(nPreviousOpStatValue, pStatListEx, opStatData.nStat);
```

Comment:
I think, this should be

nAccumulatedValue += ComputeStatPercentage(nPreviousOpStatValue, pStatListEx, opStatData.nStat);

Can you double check?